src/ListTabulator/index.ts

```diff
@@ -735,64 +928,98 @@ export default class ListTabulator {
       `.${DefaultListCssClasses.itemChildren}`
     );
 
-    this.caret.save();
+    const restore = save();
 
     /**
      * If prev item has child items, just append current to them
+     * Else render new child wrapper for previous item
      */
     if (prevItemChildrenList) {
       /**
-       * CurrentItem would not be removed soon (it should be cleared content and checkbox would be removed)
-       * after that elements with child items would be moveable too
+       * Previous item would be appended with current item and it's sublists
+       * After that sublists would be moved one level back
+       */
+      prevItemChildrenList.appendChild(currentItem);
+
+      /**
+       * Get current item child wrapper
+       */
+      const currentItemChildWrapper = currentItem.querySelector(`.${DefaultListCssClasses.itemChildren}`) as HTMLElement;
+
+      /**
+       * If current item has no child wrapper, than it has no childs
+       * So it is completely moved to new nesting level
        */
-      currentItem.remove();
-      const newSublistItem = this.list!.renderItem(this.list!.getItemContent(currentItem), {checked: false});
-      prevItemChildrenList.appendChild(newSublistItem);
+      if (currentItemChildWrapper === null) {
+        return;
+      }
+
+      /**
+       * Get all current item child to be moved to previous nesting level
+       */
+      const currentItemChildrenList = this.getChildItems(currentItemChildWrapper);
+
+      /**
+       * Move current item sublists one level back
+       */
+      if (currentItemChildrenList !== null) {
+        currentItemChildrenList.forEach((child) => {
+          prevItemChildrenList.appendChild(child);
+        })
+      }
     } else {
+      const prevItemChildrenListWrapper = this.renderer!.renderWrapper(1);
+
       /**
-       * CurrentItem would not be removed soon (it should be cleared content and checkbox would be removed)
-       * after that elements with child items would be moveable too
+       * Previous item would be appended with current item and it's sublists
+       * After that sublists would be moved one level back
        */
-      currentItem.remove();
+      prevItemChildrenListWrapper.appendChild(currentItem);
+
       /**
-       * If prev item has no child items
-       * - Create and append children wrapper to the previous item
-       * - Append current item to it
+       * Get current item child wrapper
        */
-      const sublistWrapper = this.list!.renderWrapper(1);
-      const newSublistItem = this.list!.renderItem(this.list!.getItemContent(currentItem), {checked: false});
+      const currentItemChildWrapper = currentItem.querySelector(`.${DefaultListCssClasses.itemChildren}`);
 
-      sublistWrapper.appendChild(newSublistItem);
+      /**
+       * Get all current item child to be moved to previous nesting level
+       */
+      const currentItemChildrenList = Array.from(currentItemChildWrapper?.children ?? []).filter(child =>
```

Comment:
this logic seems unclear for me. Why filter is needed? 

Also, this code fragment duplicates a fragment above. That means that it can be moved to util


<img width="609" alt="image" src="https://github.com/user-attachments/assets/16221961-be54-4e54-8bed-b22acdaaad5c">